### PR TITLE
Add DataImportCron ImageStream tag support

### PR DESCRIPTION
**Pull request by arnongilboa**
Signed-off-by: Arnon Gilboa <agilboa@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
Add DataImportCron ImageStream tag support
```

### CODE REVIEW

File: pkg/controller/dataimportcron-controller.go [hunk 1]
```diff
@@ -138,30 +139,61 @@ func (r *DataImportCronReconciler) initCron(ctx context.Context, dataImportCron
...
 }
 
-func (r *DataImportCronReconciler) getImageStream(ctx context.Context, imageStreamName, imageStreamNamespace string) (*imagev1.ImageStream, error) {
+func (r *DataImportCronReconciler) getImageStream(ctx context.Context, imageStreamName, imageStreamNamespace string) (*imagev1.ImageStream, string, error) {
 	if imageStreamName == "" || imageStreamNamespace == "" {
-		return nil, errors.Errorf("Missing imagestream name or namespace")
+		return nil, "", errors.Errorf("Missing imagestream name or namespace")
 	}
 	imageStream := &imagev1.ImageStream{}
+	isName, isTag, err := splitImageStreamName(imageStreamName)
```
Comment (awels) [inline pkg/controller/dataimportcron-controller.go:147]: I think the naming is throwing me off here, isName, and isTag I keep thinking why is this not a bool for the 'isSomething' naming convention, but it is actually imageStreamName, and imageStreamTag. maybe not shorten it in this case to make it clearer? Or just name and tag, since we are in the getImageStream function.

File: pkg/controller/dataimportcron-controller_test.go [hunk 2]
```diff
@@ -353,7 +354,8 @@ func createDataImportCronReconciler(objects ...runtime.Object) *DataImportCronRe
 
 func newDataImportCronWithImageStream(name string) *cdiv1.DataImportCron {
```
Comment (awels) [inline pkg/controller/dataimportcron-controller_test.go:355]: Can we add a test that tests the permutations of imagestreams, and invalid ones?
imagestream
imagestream:
imagestream:tag
imagestream:tag:label

File: pkg/controller/dataimportcron-controller_test.go [hunk 3]
```diff
@@ -369,6 +371,10 @@ func newImageStream(name string) *imagev1.ImageStream {
 		Status: imagev1.ImageStreamStatus{
 			Tags: []imagev1.NamedTagEventList{
 				{
+					Tag: "nosuch",
```
Comment (awels) [inline pkg/controller/dataimportcron-controller_test.go:374]: Maybe also test the negative case where we have a tag that is not in the status.tags? Also are we still testing the no tag case?

Comment (arnongilboa) [reply pkg/controller/dataimportcron-controller_test.go]: added

File: pkg/controller/dataimportcron-controller_test.go (hunk 2 above)
Comment (arnongilboa) [reply pkg/controller/dataimportcron-controller_test.go]: added

File: pkg/controller/dataimportcron-controller.go (hunk 1 above)
Comment (arnongilboa) [reply pkg/controller/dataimportcron-controller.go]: fixed
